executor/set.go

```diff
@@ -108,6 +109,10 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
 	if sysVar == nil {
 		return variable.ErrUnknownSystemVar.GenWithStackByArgs(name)
 	}
+	// The 'baseline evolution' only work in the CI environment before the feature is GA.
+	if sysVar.Name == "tidb_evolve_plan_baselines" && !config.CheckTableBeforeDrop {
```

Comment:
We should check the target value is `on` before reporting error, otherwise, if the users have already enabled this feature, they cannot disable the feature anymore.